Detections/MultipleDataSources/ExtensiveAWSComputeResurseDeployements.yaml

```diff
@@ -0,0 +1,54 @@
+id: 9e457dc4-81f0-4d25-bc37-a5fa4a17946a
+name: Extensive AWS Compute Resurse Deployements
+description: |
+  'This detection focused on Suspicious deployment of AWS EC2 resource (virtual machine) scale sets was detected. This behavior might indicate that the threat actor is deploying computing resources for cryptocurrency mining activities.This detection centers around identifying suspicious instances of AWS EC2 resource deployment, particularly scale sets. Such behavior raises concerns of potential threat actor involvement, potentially indicative of efforts to deploy computing resources for the purpose of cryptocurrency mining activities.
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AWS
+    dataTypes:
+      - AWSCloudTrail
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  -  Impact
+relevantTechniques:
+  - T1496
+query: |
+  // Retrieve AWS CloudTrail events generated within the last day
+  AWSCloudTrail
+  | where TimeGenerated >= ago(1d)
+  // Filter events related to instance creation
+  | where EventName =~ "RunInstances"
+  // Exclude events with error messages
+  | where isempty(ErrorMessage)
+  // Extract the event source type
+  | extend EventSourceSplit = split(EventSource, ".")
+  | extend Type = tostring(EventSourceSplit[0])
+  // Extract instance-related details from the event data
+  | extend instance = tostring(parse_json(RequestParameters).instanceType),platform = tostring(parse_json(ResponseElements).instancesSet.items[0].platform)
+  // Determine the operating system platform
+  | extend OSplatform = iff(isempty(platform), tostring("Linux"), platform),CPU = tostring(parse_json(ResponseElements).instancesSet.items[0].cpuOptions),core = toint(parse_json(ResponseElements).instancesSet.items[0].cpuOptions.coreCount),corThread = toint(parse_json(ResponseElements).instancesSet.items[0].cpuOptions.threadsPerCore),InstanceId = tostring(parse_json(ResponseElements).instancesSet.items[0].instanceId)
+  // Calculate the total compute based on core and thread counts
+  | extend totalCorecompute = core * corThread
+  // Filter out instances with empty core values

```

Comment:
How do you have instances without CPU cores?

Reply:
there are some instance where we are not able to find the core numbers, this could be telemetry gap, to reduce this kind of noise I added this conditions. I just edit it one line above in query.